### DISCUSSION TITLE
Need advice: Synology vs Unraid for Docker SageTV

### DISCUSSION THREAD

**Discussion by Flying-Shawn** [OpenSageTV/Community#103] [Q&A]
Hello fellow SageTV holdouts…  I’m preparing to upgrade/overhaul my ancient SageTV infrastructure and I’d greatly appreciate your input!

### **Current setup:**
A dedicated Windows 8 HTPC running SageTV 9.2.2 with a 4-core Sandy Bridge Intel i5-2500K, 24GB of RAM, and an SSD used as recording cache.
    +
Virtualized Unraid on an ESXI 6.7 server with a 4-core+hyperthreading Ivy Bridge Xeon E3-1280 V2, 32GB of RAM (8GB dedicated to Unraid), 30TB of storage (4x 12TB + 1 6TB WD 5400RPM hard drives including dual parity and an SSD cache, controlled via a passthrough’d LSI-2008) hosting my SageTV recording repository (I manually move the recordings from the HTPC cache drive every few months), all connected to the HTPC via gigabyte LAN.
    +
Four tuners: two Colossus cards with USB-UIRTs connected to my Verizon FiOS STBs and an HDHomeRun Dual.
    +
All watched via two HD300 Extenders

### **Reasons for needing the upgrade:**

- Browsing the SageTV Recording interface has been DOG SLOW ever since I migrated my recordings over to the Unraid box, especially when using the HD300s, which is causing me to use Sage less-and-less often.
- The current HTPC is old and the Windows install isn’t entirely stable
- It seems what little development is still happening on Sage is all focused on the Linux and Docker versions

### **New setup plan(s):**
A virtualized Windows 8.1 SageTV network encoder so I can still use my Colossus tuners/access STB channels (this is already set up and running with one of the cards)
    +
The two HD300s, plus a couple of new local clients and probably 2-3 remote Placeshifters/VPN’d clients (undecided on client hardware)
    +
_----EITHER----_

- Option A.  Docker SageTV on the existing Unraid server (which has recently been upgraded to 10G ethernet to connect to my new Ubiquiti router/switches)
- Option B.  Docker SageTV on a DSM 7.2 Synology DS1815+ that I repaired+upgraded to 16GB RAM after inheriting it broken from a family member.  I’ve set it up with a 31TB array of 18TB Seagate 7200RPM NAS drives in a fast RAID 10 configuration.  This NAS has four gigabyte ethernet ports, which I can either connect to the Ubiquiti switch individually with SMB Multichannel, aggregate together as a bonded link, or some combo thereof.
- Option C.  SageTV Server installed on a newer/faster low-cost third platform (mini-PC?  Raspberry Pi?  Something else?) recording directly to either the NAS or the existing Unraid VM over the network.

### **Goals/concerns:**
- Which option is going to give me a faster/better/smoother experience when using SageTV from my extenders/clients?  Should I prioritize the higher compute resources in the Unraid box or the much-faster storage array in the NAS?
- I’m probably going to be setting up Jellyfin in a docker on either the NAS or Unraid VM as a way of accessing/streaming my ripped 4K blu-ray library (and maybe as an easier way of sharing my Sage recordings with family remotely vs trying to set them up with Sage clients): would it be best to put SageTV+Recordings on one platform and Jellyfin+Blu-rays on the other to spread out resources?  If so, which goes where?
- Option C might offer the fastest compute hardware for tasks like transcoding vs the aging Xeon Unraid server or the slow Atom chip in the NAS, but I’m worried it’d have the same interface lags over the network as my current setup because it’s not local to the storage.
- And, yes, I’ll say it now… I know the “best” option would be to overhaul the virtual server with better mobo/cpu/ram, allocate more resources to the Unraid VM, and install all my Dockers together there, but that’d also be a much higher-cost solution.

Any thoughts?  It feels weird to be considering such a major overhaul with a “dying” platform such as SageTV, but even after all these years there _still_ doesn’t seem to be anything out there that beats it!

**Comment by wayner9** [comment]
Since you aren't getting much response I will go - I have been using unRAID for years.  My first server I built about 5-6 years ago.  Last year I built another server and have them both running.  The speed is pretty good, the best that I have seen for SageTV.  Maybe your virtualized unRAID is what is causing issues.  SageTV runs better when the docker appdata folder is on an SSD/NVME, which is how I have it.

I am also running Plex and Jellyfin dockers.  Jellyfin I just installed yesterday.  They both work fine pointed to the same media folders as SageTV, although I don't have them pointed at my TV recordings folder.  I think they both want TV shows in "named" folders rather than all jammed into one folder.  They are better for playback when I add media files to my folder.  Often SageTV doesn't like the timeline for files and playback is kind of hosed.

**Reply by Flying-Shawn** [reply]
Thanks wayner!  To be clear, my current laggy setup is a hardware Windows SageTV install accessing unRAID over gigabit LAN rather than any sort of docker SageTV running within the virtualized unRAID (which would be option A above).  That said, if I decide to go that route, your comment is a good reminder to make sure I run the Dockers off an SSD (NVME sadly isn't an option in that box) rather than allowing them to be installed to the HDD array.

**Reply by wayner9** [reply]
I wouldn't worry about network speed to much as I am pretty sure that the HD300s only 100M ethernet ports and that is plenty of speed for delivering SageTV.  I think the speed of the server and accessing stuff like Fanart is the real bottleneck.  But having a fast switch is helpful if you have a lot going on in your house.  I have Ubiquiti hardware for my router (USG), WAPs and a couple of switches, but most of my switches are unmanaged 1G switches.

I don't know that you have to worry about spreading resources across multiple drives or devices.

**Reply by trk2** [reply]
Hey Wayner, do you use the Media Format Parser plugin?  It will most likely resolve your timeline issues on imports.
